### PR TITLE
Make compressing serializers not close argument streams

### PR DESCRIPTION
**Pull request by groldan**
Signed-off-by: Gabriel Roldan <groldan@boundlessgeo.com>

### CODE REVIEW

File: src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java [hunk 1]
```diff
@@ -77,10 +92,10 @@ public void write(RevObject o, OutputStream out) throws IOException {
...
         LZ4Compressor compressor = lz4factory.fastCompressor();
         Checksum checksum = newChecksum();
-        try (LZ4BlockOutputStream cout = new LZ4BlockOutputStream(out, blockSize, compressor,
-                checksum, false)) {
-            factory.write(o, cout);
-        }
+        LZ4BlockOutputStream cout = new LZ4BlockOutputStream(out, blockSize, compressor, checksum,
+                true);
+        factory.write(o, cout);
+        cout.finish();// same as close but not closing the wrapped stream
```
Comment (DBlasby) [inline src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java]: should this be in a finalize?

File: src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java [hunk 2]
```diff
@@ -51,13 +57,22 @@ public LZ4SerializationFactory(final ObjectSerializingFactory factory) {
...
+    }
+
     @Override
     public RevObject read(ObjectId id, InputStream in) throws IOException {
         LZ4FastDecompressor decompressor = lz4factory.fastDecompressor();
         Checksum checksum = newChecksum();
-        try (LZ4BlockInputStream cin = new LZ4BlockInputStream(in, decompressor, checksum)) {
-            return factory.read(id, cin);
-        }
+        LZ4BlockInputStream cin = new LZ4BlockInputStream(in, decompressor, checksum);
```
Comment (DBlasby) [inline src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java]: These seem to be created but not closed - is this correct?

I'm not seeing anything in the factory.read() interface to say what's the expected behaviour.

File: src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java (hunk 1 above)
Comment (groldan) [reply src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java]: I don't think so, if factory.write(o, cout) throws an exception then it's propagated and nothing can really be done so there's no point in forcing to flush the compression buffer

File: src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java (hunk 2 above)
Comment (groldan) [reply src/core/src/main/java/org/locationtech/geogig/storage/datastream/LZ4SerializationFactory.java]: It's correct, the uncompressing stream decorator shall not close the argument stream

**Pull request merged**